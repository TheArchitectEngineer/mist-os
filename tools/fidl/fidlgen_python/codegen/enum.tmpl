{{/*
// Copyright 2025 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
*/}}

{{/*
// TODO(https://fxbug.dev/394421154): This should really be a class that inherits from enum.IntFlag,
// but doing so requires creating a new GIDL backend because the fuchsia-controller backend
// generates tests that use ints for the value of bits types.
*/}}
{{ define "EnumDeclaration" -}}
{{ .PythonName }} = enum.IntEnum("{{ .PythonName }}", {
    {{- if .DocComments }}
    "__doc__": """
        {{- range .DocComments }}
{{ . | escapeQuotes | trimSpace | indentNonEmpty8 }}
        {{- end -}}
        """,
    {{- end }}
    "__fidl_kind__": "enum",
    "__fidl_type__": "{{ .PythonName }}",
    "__fidl_raw_type__": "{{ .Name }}",
    "__strict__": {{ if .Strictness }}True{{ else }}False{{ end }},
    {{ range .PythonMembers -}}
    "{{ .PythonName }}": {{ .PythonValue }},
    {{ end }}
    {{- if or .Empty (not .HasZero) -}}
    "EMPTY__": 0,
    {{ end -}}
    "make_default": classmethod(lambda cls: cls(0)),
})


{{ end }}