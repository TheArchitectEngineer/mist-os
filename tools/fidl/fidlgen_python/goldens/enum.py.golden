# WARNING: This file is machine generated by fidlgen.

from __future__ import annotations

from dataclasses import dataclass

# TODO(https://fxbug.dev/346628306): Enable type checking here once fidl_codec has stubs
from fidl_codec import add_ir_path, encode_fidl_object  # type: ignore
import sys
import typing
import enum

import fidl

_ir_path = fidl._library.get_fidl_ir_map()["test.enum"]
add_ir_path(_ir_path)


MyEmptyFlexibleEnum = enum.IntEnum(
    "MyEmptyFlexibleEnum",
    {
        "__fidl_kind__": "enum",
        "__fidl_type__": "MyEmptyFlexibleEnum",
        "__fidl_raw_type__": "test.enum/MyEmptyFlexibleEnum",
        "__strict__": False,
        "EMPTY__": 0,
        "make_default": classmethod(lambda cls: cls(0)),
    },
)


MyFlexibleEnum = enum.IntEnum(
    "MyFlexibleEnum",
    {
        "__fidl_kind__": "enum",
        "__fidl_type__": "MyFlexibleEnum",
        "__fidl_raw_type__": "test.enum/MyFlexibleEnum",
        "__strict__": False,
        "FOO": 1,
        "BAR": 2,
        "EMPTY__": 0,
        "make_default": classmethod(lambda cls: cls(0)),
    },
)


MyFlexibleEnumWithCustomUnknown = enum.IntEnum(
    "MyFlexibleEnumWithCustomUnknown",
    {
        "__fidl_kind__": "enum",
        "__fidl_type__": "MyFlexibleEnumWithCustomUnknown",
        "__fidl_raw_type__": "test.enum/MyFlexibleEnumWithCustomUnknown",
        "__strict__": False,
        "FOO": 1,
        "BAR": 2,
        "CUSTOM_UNKNOWN": 3,
        "EMPTY__": 0,
        "make_default": classmethod(lambda cls: cls(0)),
    },
)


MyStrictEnum = enum.IntEnum(
    "MyStrictEnum",
    {
        "__fidl_kind__": "enum",
        "__fidl_type__": "MyStrictEnum",
        "__fidl_raw_type__": "test.enum/MyStrictEnum",
        "__strict__": True,
        "FOO": 1,
        "BAR": 2,
        "EMPTY__": 0,
        "make_default": classmethod(lambda cls: cls(0)),
    },
)
